src/main/java/net/countercraft/movecraft/combat/movecraftcombat/directors/CannonDirectorManager.java

```diff
@@ -122,11 +123,29 @@ private void processDirectors() {
                 if (!(tnt.getVelocity().lengthSquared() > 0.35) || tracking.containsKey(tnt)) {
                     continue;
                 }
-                Craft c = CraftManager.getInstance().fastNearestCraftToLoc(tnt.getLocation());
+
+                Craft c;
+
+                if(Config.EnableTNTTracking) {
+                    if(!tnt.hasMetadata("MCC-Sender")) {
+                        continue;
```

Comment:
If the TNT is not tagged with a sender, then the default behavior should match that of the current method.